lib/components/delete_profile_dialog.dart

```diff
@@ -0,0 +1,40 @@
+import 'package:provider/provider.dart';
+import 'package:flutter/material.dart';
+import 'package:watermelon/model/profiles.dart';
+import '../model/state.dart';
+
+Future<void> deleteProfileDialog(BuildContext context, UserProfile profile) {
+  return showDialog<void>(
+    context: context,
+    builder: (BuildContext context) {
+      var appState = context.watch<AppState>();
+      return AlertDialog(
+        title: const Text('Delete profile'),
+        content: const Text(
+          "This will delete the profile from your device. Please ensure you have backed up the private key.",
```

Comment:
To make a little more explicit.

```suggestion
          "This will delete the profile from your device. Please ensure you have backed up the private key before deleting. This key cannot be recovered once deleted.",
```